File: oop/python/lista01/Ex1.py
from random import randrange

from KeyboardInput import KeyboardInput

class Ex1:
    def sqrt(self, n) -> float:
        previousAttempt = randrange(1, n)
        currentAttempt = (previousAttempt + n/previousAttempt) / 2

        while previousAttempt - currentAttempt > 0.00000001:
            previousAttempt = currentAttempt
            currentAttempt = (previousAttempt + n/previousAttempt) / 2
        return currentAttempt

if __name__ == "__main__":
    squareRooter = Ex1()
    keyboardInput = KeyboardInput()

    n = keyboardInput.readPositiveInt("Provide a number: ")
    print("Square root: {}".format(squareRooter.sqrt(n)))

File: oop/lista01_java/Ex01/Ex1.java
package Lista01.Ex01;

import Lista01.KeyboardInput;

import java.util.Random;

public class Ex1 {
    private final Random rand = new Random();

    public Double sqrt(Double n) {
        Double attempt = (double) rand.nextInt(n.intValue()) + 1;
        while (attempt - (attempt = (attempt + n/attempt)/2) > 0.00000001);
        return attempt;
    }

    public static void main(String[] args) {
        Ex1 squareRooter = new Ex1();
        KeyboardInput keyboardInput = new KeyboardInput(System.in);

        System.out.println("Provide a number: ");
        Double input = keyboardInput.readDouble();

        System.out.println("Square root: " + squareRooter.sqrt(input));
    }
}

